frontend/packages/dev-console/src/actions/providers.ts

```diff
@@ -16,3 +24,71 @@ export const useEditImportActionProvider = (resource: K8sResourceKind) => {
 
   return [editImportAction, !inFlight, undefined];
 };
+
+export const useTopologyGraphActionProvider = (element: GraphElement) => {
+  const [namespace] = useActiveNamespace();
+  const [
+    addActionExtensions,
+    addActionExtensionsResolved,
+    allAddActionsDisabled,
+  ] = useAddActionExtensions();
+  const [
+    filteredAddActionExtensions,
+    filteredAddActionExtensionsLoaded,
+  ] = useAccessFilterExtensions(
+    namespace,
+    filterNamespaceScopedUrl(namespace, addActionExtensions),
+  );
+  const loaded =
+    !allAddActionsDisabled && addActionExtensionsResolved && filteredAddActionExtensionsLoaded;
+
+  return React.useMemo(() => {
+    if (element.getKind() === ModelKind.graph) {
+      const actions =
+        loaded &&
+        filteredAddActionExtensions
+          .filter((a) => !!a.properties.action)
+          .map((a) => contextAction(a, namespace));
+      return [actions, loaded, undefined];
+    }
+    return [[], true, undefined];
+  }, [element, loaded, filteredAddActionExtensions, namespace]);
+};
+
+export const useTopologyApplicationActionProvider = (element: GraphElement) => {
+  const [namespace] = useActiveNamespace();
+  const [
+    addActionExtensions,
+    addActionExtensionsResolved,
+    allAddActionsDisabled,
+  ] = useAddActionExtensions();
+  const [
+    filteredAddActionExtensions,
+    filteredAddActionExtensionsLoaded,
+  ] = useAccessFilterExtensions(
+    namespace,
+    filterNamespaceScopedUrl(namespace, addActionExtensions),
+  );
+  const loaded =
+    !allAddActionsDisabled && addActionExtensionsResolved && filteredAddActionExtensionsLoaded;
+
+  return React.useMemo(() => {
+    if (element.getType() === TYPE_APPLICATION_GROUP) {
+      const appName = element.getLabel();
+      const appData: TopologyApplicationObject = {
+        id: element.getId(),
+        name: appName,
+        resources: element.getData().groupResources,
+      };
+
+      const actions = loaded && [
+        ...filteredAddActionExtensions
+          .filter((a) => a.properties.action?.isGroupable)
+          .map((a) => contextAction(a, namespace, appName)),
+        applicationDeleteAction(appData),
```

Comment:
delete action should come before all other menu items as it is on master

![image](https://user-images.githubusercontent.com/9278015/135081599-29474717-da73-4d33-92a2-cdb777ab2e26.png)